library/std/src/sys_common/backtrace.rs

```diff
@@ -213,3 +214,14 @@ pub fn output_filename(
     }
     fmt::Display::fmt(&file.display(), fmt)
 }
+
+#[cfg(all(target_vendor = "fortanix", target_env = "sgx"))]
+pub fn set_image_base() {
+    let image_base = crate::os::fortanix_sgx::mem::image_base();
+    backtrace_rs::set_image_base(crate::ptr::from_exposed_addr_mut(image_base as _));
```

Comment:
Is this API implying that the backtrace generation is going to do more than math with the pointer value here? i.e., read/write from it? It seems odd to me to call from_exposed_addr here, vs. invalid_mut, which feels like it more appropriately describes the opaque nature of this pointer and theoretically hints that no one should read from it.

Reply:
No, backtrace is only using this for adjusting frame IPs. Having read the docs in https://doc.rust-lang.org/nightly/std/ptr/index.html I was under the impression that doing pointer arithmetic also requires strict provenance.

Reply:
See https://github.com/rust-lang/backtrace-rs/blob/master/src/backtrace/libunwind.rs#L38-L55

Reply:
I would not expect arithmetic with non-unsafe functions to ever require provenance. It looks like that code is operating on `usize` even, so it's not even doing pointer arithmetic I think.